lib/puppet/functions/length.rb

```diff
@@ -0,0 +1,16 @@
+#A function to eventually replace the old size() function for stdlib - The original size function did not handle Puppets new type capabilities, so this function is a Puppet 4 compatible solution. 
+Puppet::Functions.create_function(:length) do
+	dispatch :length do
+		param 'Any', :value
```

Comment:
Using the proper type declaration here, using `Variant`, will avoid the need for the `raise` in L12 below.

Using the proper type declarations here, using different dispatches for Strings, and others, will avoid the need for the `if` chain in L7.